src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        `All sides of a triangle must be greater than zero. ` +
+          `Given sides: a=${a}, b=${b}, c=${c}`,
+      );
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(
+        `The sides ${a}, ${b}, and ${c} cannot form a valid triangle`,
+      );
+    }
+  }
+
+  public getArea = (): number => {
+    const s: number = (this.a + this.b + this.c) / 2;
+    const triangleArea: number = Math.sqrt(
+      s * (s - this.a) * (s - this.b) * (s - this.c),
+    );
+
+    return Math.round(triangleArea * 100) / 100;
```

Comment:
According to the checklist, the `getArea` function should not accept any parameters since all necessary values are available as object fields. Please remove the parameter list from the `getArea` method declaration.